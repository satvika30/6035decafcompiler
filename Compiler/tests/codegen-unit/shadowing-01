class Program
{
	int A;
    void changeA()
    {
    	int A;
    	A = 20;
    	callout("printf","A = %d\n", A);
    	if(true) {
    		A = 12;
    		callout("printf", "A = %d\n", A);
    	}
    }


    void main()
    {
    	A=0;
		callout("printf","A = %d\n", A);
		changeA();
		callout("printf","A = %d\n", A);
     
    }

}

