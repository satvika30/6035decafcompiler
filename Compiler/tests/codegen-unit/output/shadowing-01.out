A = 0
A = 20
A = 12
A = 0
